### PR TITLE
feat: show the real file name when displaying git diff result

### PR DESCRIPTION
**Pull request by LucianLiu6**
### Before my fix
When using `icdiff` to display the git diff result, the file path is not unrecognizable for human.

Just like this
filename-unrecognizable
Let's take a look at the file paths which are:
1. ChangeLog
    * `/var/folders/h1/jvqp6h_50dvfnjz56cv8ssq4067fyr/T//zGL6ka_ChangeLog`
    * `/var/folders/h1/jvqp6h_50dvfnjz56cv8ssq4067fyr/T//1s1fqa_ChangeLog`
2. MANIFEST.in
    * `/var/folders/h1/jvqp6h_50dvfnjz56cv8ssq4067fyr/T//LeC2ab_MANIFEST.in`
    * `/var/folders/h1/jvqp6h_50dvfnjz56cv8ssq4067fyr/T//12boab_MANIFEST.in`

If your file located in the root directory, you may can find the file name according to the tail.  
But If your file located in the subdirectory, the subdirectory will be missed, you will only get the file name.   
Let's say the file path is `${project_root}/a/b/c/file.txt`, the file path displayed by icdiff will be like this `/var/folders/h1/jvqp6h_50dvfnjz56cv8ssq4067fyr/T//12boab_file.txt`

### After my fix
The file path will be easy to read, and the dir-path of files which located in subdirectory will not miss.
filename-corrected

### How I fix it
* Add a arg which named `--is-git-diff` when call `git-icdiff`
* Change the header to env variable `$BASE`

PS: see more details about `$BASE` [here](https://git-scm.com/docs/git-difftool#Documentation/git-difftool.txt---extcmdltcommandgt)

### CODE REVIEW

**Comment by LucianLiu6** [comment]
> Looks good! Could you add a test?

Ok, I will add one tomorrow

File: test.sh [hunk 1]
```diff
@@ -120,6 +144,7 @@ check_gold gold-subcolors-bad-color tests/input-{1,2}.txt --cols=80 --color-map=
 check_gold gold-subcolors-bad-cat tests/input-{1,2}.txt --cols=80 --color-map='chnge:magenta,description:cyan_bold'
 check_gold gold-subcolors-bad-fmt tests/input-{1,2}.txt --cols=80 --color-map='change:magenta:gold,description:cyan_bold'
 check_gold gold-bad-encoding.txt tests/input-{1,2}.txt --encoding=nonexistend_encoding
+check_git_diff gitdiff-only-newlines.txt 4e862056296aa810a46caaf2ac1b0feb82c5e330~1 4e862056296aa810a46caaf2ac1b0feb82c5e330
```
Comment (jeffkaufman) [inline test.sh]: optional nit: you could use shorter hashes here, like 4e86205629~1 and 4e86205629

Comment (LucianLiu6) [reply test.sh]: I found that Travis CI only clone **depth 50** of repository...

So I can't use the very old hashes in icdiff repo to test the git-icdiff result

I'm thinking about create a new repo which named `icdiff-git-tests` to place the git-icdiff test cases

Comment (LucianLiu6) [reply test.sh]: Here is the screenshot of Travis Log

屏幕快照 2019-07-17 下午10 40 16

Comment (jeffkaufman) [reply test.sh]: Can you turn off the `--depth` flag with `depth: false`?  https://docs.travis-ci.com/user/customizing-the-build/#git-clone-depth

If that doesn't work you could have the test do something like:

```
git clone https://github.com/jeffkaufman/icdiff.git tmp
cd tmp
[run git-icdiff]
```

Comment (LucianLiu6) [reply test.sh]: Good Idea, I'll try the `depth: false`.

btw, I'm confused with how `git icdiff xx xx` works.  
It works very fine on my Mac, but the Travis CI complain that:
```shell
git: 'icdiff' is not a git command.
```
I have added the `ic-diff` shell script into PATH before I call `git icdiff xx xx`

Do you have any suggestions to fix this Travis error?

Comment (jeffkaufman) [reply test.sh]: I think the problem is your `export PATH=$PATH:"$(pwd)/git-icdiff"`  Instead you want `export PATH=$PATH:"$(pwd)".  `PATH` says what directories to check for executables, and isn't a listing of executables directly.

Comment (LucianLiu6) [reply test.sh]: Oh, you must be right!

I made such a stupid mistake 😂

**Comment by LucianLiu6** [comment]
OK, tests all passed finally.

**Comment by jeffkaufman** [comment]
thanks a lot for the patch, and for adding the first git-icdiff test!

**Comment by LucianLiu6** [comment]
I will find some other time to add more test cases of git-icdiff such as `only-deletes`, `both-newlines-deletes`, `binary-files` and so on...

**Comment by ViktorQvarfordt** [comment]
When will this be in a release?

**Comment by jeffkaufman** [comment]
This is now released in 1.9.5

**Pull request merged**